### PR TITLE
Clear internal stream map on connection close.

### LINKED ISSUE #12135: StreamId Map should be cleared when connection is stopped
Currently we don't clear the internal dictionary for mapping stream ids to active streams when the client disconnects. It's a silent issue because we will override any stale values with new streams, but regardless, we should be clearing out the map on connection close.

### PR DESCRIPTION
**Pull request by mikaelm12**
Fixes: https://github.com/aspnet/AspNetCore/issues/12135

Clears our internal stream map on close.

### CODE REVIEW

**Comment by mikaelm12** [comment]
Needs to be updated based on conversation in #12135, going to make a draft for now

**Review by jkotalik — APPROVED** [review]

File: src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java [hunk 1]
```diff
@@ -515,6 +515,70 @@ public void checkStreamUploadThroughSendWithArgs() {
...
+        assertEquals("{\"type\":2,\"invocationId\":\"1\",\"item\":\"FirstItem\"}\u001E", messages[3]);
+        assertEquals("{\"type\":2,\"invocationId\":\"2\",\"item\":\"SecondItem\"}\u001E", messages[4]);
+
+
+        assertEquals(2, hubConnection.getStreamMap().size());
+        assertTrue(hubConnection.getStreamMap().keySet().contains("1"));
+        assertTrue(hubConnection.getStreamMap().keySet().contains("2"));
+
+        // Verify that we clear the entry from the stream map after we clear the first stream.
+        stream.onComplete();
```
Comment (halter73) [inline src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java]: Do we verify the streamMap entry gets cleared when sendHubMessage completes with an error?

**Review by halter73 — APPROVED** [review]

File: src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java (hunk 1 above)
Comment (mikaelm12) [reply src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java]: Was just thinking this as I was headed to the restroom. I'll complete the second stream with an error and verify.

File: src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java [hunk 2]
```diff
@@ -109,6 +109,11 @@ void setTickRate(long tickRateInMilliseconds) {
         this.tickRate = tickRateInMilliseconds;
     }
 
+    // For testing purposes
+    Map<String,Observable> getStreamMap() {
```
Comment (BrennanConroy) [inline src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java]: nit: space

**Review by BrennanConroy — APPROVED** [review]

**Comment by mikaelm12** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=270213

**Pull request merged**